### PR TITLE
feat(gitlab): added gitlab repository support

### PR DESCRIPTION
**Pull request by ochrstn**
👋 

Support for gitlab.com or self-hosted instances:

```sh
# self-hosted
docker run --rm -it gitopscli:latest deploy --file values.yaml --values "{ app.image.tag: 2.5.0 }" --username GitOps --password 1GdNwR2QDit6xjwGxkjC --organisation "not needed" --repository-name 301 --git-provider-url "https://gitlab.yourorg.com" --create-pr --auto-merge
```

```sh
# gitlab.com
docker run --rm -it gitopscli:latest deploy --file values.yaml --values "{ app.image.tag: 2.5.0 }" --username GitOps --password JzKPDUmNvbvmohNypmb4 --organisation "not needed" --repository-name 24029565 --git-provider gitlab --create-pr --auto-merge
```

To authenticate you have to create a  [personal access token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#creating-a-personal-access-token). Username and password then correspond to the token name and the access token itself.

As repository *name* I decided to use the Gitlab Project ID which is unique and immutable for a repository on the whole gitlab server and the organization name is not really needed.

Happy for any feedback ❤️

### CODE REVIEW

**Comment by christiansiegel** [comment]
Thanks for the contribution! :heart: 

Maybe it would make sense to add a `--project-id` parameter as alternative to `--organisation` and `--repository-name` for Gitlab? I guess this would require to make both `--organisation` and `--repository-name` optional in `cliparser.py`. Instead, there could be a simple check that either org/repo or project-id parameters are given on this level (like here: https://github.com/baloise/gitopscli/blob/master/gitopscli/cliparser.py#L268). And finally an additional check in `GitRepoApiFactory` just before passing it to your adapter.

**Comment by ochrstn** [comment]
> Maybe it would make sense to add a `--project-id` parameter as alternative to `--organisation` and `--repository-name` for Gitlab? I guess this would require to make both `--organisation` and `--repository-name` optional in `cliparser.py`. Instead, there could be a simple check that either org/repo or project-id parameters are given on this level (like here: https://github.com/baloise/gitopscli/blob/master/gitopscli/cliparser.py#L268). And finally an additional check in `GitRepoApiFactory` just before passing it to your adapter.

Sounds reasonable. I will try to propose a solution where both is possible, because with Gitlab it is also possible to search a repository via organization (aka group) and repository name.

**Comment by ochrstn** [comment]
> > Maybe it would make sense to add a `--project-id` parameter as alternative to `--organisation` and `--repository-name` for Gitlab? I guess this would require to make both `--organisation` and `--repository-name` optional in `cliparser.py`. Instead, there could be a simple check that either org/repo or project-id parameters are given on this level (like here: https://github.com/baloise/gitopscli/blob/master/gitopscli/cliparser.py#L268). And finally an additional check in `GitRepoApiFactory` just before passing it to your adapter.
> 
> Sounds reasonable. I will try to propose a solution where both is possible, because with Gitlab it is also possible to search a repository via organization (aka group) and repository name.

I found a way to change it easily to the same organization/repository-name way the other git adapters are using. The information is easily visible from the url (e.g. https://gitlab.com/inkscape/inkscape) while the project id is somehow more hidden (and many people are asking how to find it 😅) 

Also, I found in the codebase that `organisation` and `repository_name` are passed around in many places and I didn't felt safe enough to add a third option everywhere.

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ docker run --rm -it baloise/gitopscli --help
 For detailed installation and usage instructions, visit [https://baloise.github.io/gitopscli/](https://baloise.github.io/gitopscli/).
 
 ## Git Provider Support
-Currently, we support both BitBucket Server and GitHub.
+Currently, we support both BitBucket Server, GitHub and Gitlab.
```
Comment (joachimprinzbach) [inline README.md:34]: Remove "both" ;-)

**Review by christiansiegel — APPROVED** [review]

**Pull request merged**